indirect.h

```diff
@@ -158,9 +158,7 @@ class indirect {
     } else {
       if (std::is_copy_assignable_v<T> && !valueless_after_move() &&
           alloc_ == other.alloc_) {
-        T tmp(*other);
```

Comment:
We assign rather than copy-swap this because we just want to store an object in dynamically assigned memory not alter exception safety guarantees. basic_optional in https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p2047r7.html does the same.